src/UnleashBuilder.php

```diff
@@ -525,14 +534,28 @@ public function build(): Unleash
         $this->initializeServices($metricsHandler, $dependencyContainer);
         $this->initializeServices($variantHandler, $dependencyContainer);
 
-        return new DefaultUnleash(
-            $this->strategies,
-            $repository,
-            $registrationService,
-            $configuration,
-            $metricsHandler,
-            $variantHandler,
-        );
+        if ($this->proxyKey !== null) {
+            $configuration = $configuration->setProxyKey($this->proxyKey);
+            $proxyRepository = new DefaultUnleashProxyRepository(
+                $configuration,
+                $httpClient,
+                $requestFactory
```

Comment:
Trailing comma for consistency:

```suggestion
                $requestFactory,
```